index.d.ts

```diff
@@ -0,0 +1,208 @@
+/// <reference types="node" />
+/// <reference types="aws-lambda" />
+/// <reference types="connect" />
```

Comment:
I've been a little unsure about what this did and what the difference was compared to adding `@types/connect` as a dev-dependency. This seem a duplicate of the devDependencies.

According to the [documentation for this directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-), it simply instructs the compiler to inline that particular module in this file when compiling (at least that's how I read it). If so, the import statements below should not be necessary right? And if so, I think you'd either add the `@types/*` devDependencies + the `import` statements _OR_ you use the `<reference types="..." />` directives.

However, if I remove the `import` statements below, VS Code complains that it doesn't know about the different types previously imported. But maybe I need to prefix them with something? Or maybe I misunderstood this whole thing.

@delvedor @sqren any ideas?

Reply:
I don't understand those directives tbh. I'd always go for the devDependencies + static import option. Maybe with the exception of Node which cannot be imported as a module.

Reply:
So you think that I could just leave them all out? If I remove them I don't seem to get any errors. 

@delvedor why are you using this in fastify and elasticsearch-js?

Reply:
> @delvedor why are you using this in fastify and elasticsearch-js?

Good question, I don't know. We have always used this notation.

Reply:
![](https://media1.giphy.com/media/ToMjGpNuOksUiclTp4c/giphy.gif?cid=3640f6095c928a3c694274774906333c)

Reply:
If I install this branch as a module in a clean folder (i.e. no devDependencies), and try to run `tsc` on a file that does nothing else than imports `elastic-apm-node`, then I get these errors:

![image](https://user-images.githubusercontent.com/10602/54907840-ca00b780-4ee6-11e9-8f9e-8434127f2365.png)

I have to manually install `@types/node`, `@types/aws-lambda`, and `@types/connect` for `tsc` to stop complaining.

It seems VS Code is smarter though and will just treat all non-installed types as `any`.

Reply:
I concluded that all TypeScript developers will have `@types/node` installed already, so this isn't really a big deal. Whereas most TypeScript users of our module will probably not use the lambda API or the connect API, and as such will not have any of those dependencies installed them selfs. Requiring users to have these `@types` installed is therefore very inconvenient for the majority of our TypeScript users. So I've now inlined those types instead into our `index.d.ts` file and removed the dependency.